src/shogun2-core/shogun2-dao/src/main/java/de/terrestris/shogun2/dao/GenericHibernateDao.java

```diff
@@ -38,15 +38,45 @@ protected GenericHibernateDao(Class<E> clazz) {
 	@Autowired
 	private SessionFactory sessionFactory;
 
-	public Session getSession() {
+	/**
+	 * Obtains the current session.
+	 *
+	 * @return
+	 */
+	private Session getSession() {
 		return sessionFactory.getCurrentSession();
 	}
 
+	/**
+	 * Return the real object from the database. Returns null if the object does
+	 * not exist.
+	 *
+	 * @param id
+	 *
+	 * @see http://www.mkyong.com/hibernate/different-between-session-get-and-session-load/
+	 *
+	 * @return The object from the database or null if it does not exist
+	 */
 	public E findById(ID id) {
 		return (E) getSession().get(clazz, id);
 	}
 
 	/**
+	 * Return a proxy of the object (without hitting the database). This should
+	 * only be used if it is assumed that the object really exists and where
+	 * non-existence would be an actual error.
+	 *
+	 * @param id
+	 *
+	 * @see http://www.mkyong.com/hibernate/different-between-session-get-and-session-load/
+	 *
+	 * @return
+	 */
+	public E loadById(ID id) {
+		return (E) getSession().load(clazz, id);
```

Comment:
This will NOT throw immediately. So you will later get an exception, if you try to access it, but it does not exist. In our case, we catch this exception in the delete method of `AbstractRestController`. `findById` will simply return null, if the object does not access. But `findById` is accessing the database immediately. See hibernate docs for details :wink: